iModelJsNodeAddon/JsInterop.cpp

```diff
@@ -922,6 +931,80 @@ DbResult JsInterop::ImportSchemas(DgnDbR dgndb, bvector<Utf8String> const& schem
     return dgndb.SaveChanges();
     }
 
+//---------------------------------------------------------------------------------------
+// @bsimethod
+//---------------------------------------------------------------------------------------
+DbResult JsInterop::ConvertEC2XmlSchemas(bvector<Utf8String> const& ec2XmlStrings, bvector<Utf8String>& ec3XmlStrings, ECSchemaReadContextPtr schemaContext)
+    {
+    if (0 == ec2XmlStrings.size())
+        return BE_SQLITE_ERROR;
+
+    if (schemaContext.IsNull())
+        schemaContext = ECSchemaReadContext::CreateContext(false /*=acceptLegacyImperfectLatestCompatibleMatch*/, true /*=includeFilesWithNoVerExt*/);
+
+    JsInterop::AddFallbackSchemaLocaters(schemaContext);
+
+    bvector<bpair<SchemaKey, ECSchemaPtr>> schemaKeyPairs;
+    StringSchemaLocater locater;
+    for (Utf8String ec2XmlString : ec2XmlStrings)
+        {
+        SchemaKey key;
+        SchemaReadStatus status = ECSchema::ReadSchemaKey(ec2XmlString, key);
+        if (SchemaReadStatus::Success != status)
+            return BE_SQLITE_ERROR;
+        locater.AddSchemaString(key.GetName(), ec2XmlString);

```

Comment:
I mentioned on the locater that it should be aware of the schema keys and not just match the name, so better provide the key here, you can internally still use a map keyed on name in the locater but still remember the key, as you already have it here.
Hmm on second thought, the StringSchemaLocater probably needs to support just names as well, because that's what all our test code will be calling. It should probably have an overload which takes the key and xml, and another which takes the name and xml, but that second internally would call ReadSchemaKey() and remember it.
The locater could then expose a list of its known schema keys, which you could use in the for loop below instead of filling schemaKeyPairs. That way you only have to fill and touch the schemaKeyPairs vector once.